File: Data Structures & Algorithms/top-k-elements-in-list/submission-4.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        Map<Integer, Integer> frequencyMap = new HashMap<>();

        PriorityQueue<int[]> minHeap = new PriorityQueue<>((a, b) -> Integer.compare(a[1], b[1]));

        for (int num : nums) {
            int frequency = frequencyMap.getOrDefault(num, 0) + 1;
            frequencyMap.put(num, frequency);
        }

        for (Map.Entry<Integer, Integer> entry : frequencyMap.entrySet()) {
            minHeap.offer(new int[]{entry.getKey(), entry.getValue()});

            if (minHeap.size() > k) minHeap.poll();
        }

        return minHeap.stream().mapToInt(x -> x[0]).toArray();
    }
}

File: Data Structures & Algorithms/missing-number/submission-4.py
class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        n = len(nums)

        ret = 0
        i = 0
        while i < n:
            ret ^= i
            ret ^= nums[i]

            i += 1
        
        ret ^= n
        return ret